core/src/main/java/com/linecorp/armeria/client/endpoint/StickyEndpointSelectionStrategy.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.endpoint;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.function.ToLongFunction;
+
+import com.google.common.hash.Hashing;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.Endpoint;
+
+/**
+ * An {@link EndpointSelector} strategy which implements sticky load-balancing using
+ * user passed {@link ToLongFunction<ClientRequestContext>} as hash calculator for
+ * consistent hashing.
+ */
+public final class StickyEndpointSelectionStrategy implements EndpointSelectionStrategy {
+
+    private final ToLongFunction<ClientRequestContext> requestHasher;
+
+    /**
+     * Creates a new {@link StickyEndpointSelectionStrategy}
+     * with default hash function to hash a {@link ClientRequestContext} to a {@code long}.
```

Comment:
I don't think we need the word `default`, it's confusing me since I would expect the default to take no parameters (be provided by armeria).

How about

`with the provided function to hash a {@link ctx} to {@code l}.`

?